dvc/remote/local.py

```diff
@@ -188,7 +188,7 @@ def hardlink(self, from_info, to_info):
                     src=str(from_info), dest=str(to_info)
                 )
             )
-            return
+            return True
 
         System.hardlink(from_info, to_info)
```

Comment:
What will be returned here? Shouldn't we explicitly return False when we don't want to skip? 

Also, if there is a problem with link verification, maybe we should handle this particular use case in `_verify_link`?
When link type is hardlink and file is empty, don't set `cache_type_confirmed` but also don't raise Exception, because this particular case is special in our situation? What do you think?

Reply:
Too lazy to change this on every kind of links(they are in SSH as well). I also didn't want to spread the information of this optimization all around the codebase.

Reply:
If it'd be easier, I'd return `True` in case of explicit need of verification because `hardlink` returning True in optimization kind of looks awkward.

Reply:
@skshetry writing empty file looks awkward in our `remote/local.hardlink` :)
Also, if that is a special situation, why not override `_verfiy_link` in local?
I would think something like:

```
    def _verify_link(self, path_info, link_type):
        if self.cache_type_confirmed:
            return

        if link_type is "hardlink" and self.getsize(path_info) == 0:
            return
        super()._verify_link(path_info, link_type)
```

It's not pretty, but at least we keep it in one place. 
Having said so, it's opinionated, so I won't be pushing for this approach.

Reply:
@JIoJIaJIu @pared, still, `verify_link` is not verifying anything (I mean, it shouldn't even reach here). @pared, thanks for the suggestion, it's not as convoluted as my PR, so, I'll go with what you suggested. I'll create a separate issue to fix this *hack* later.